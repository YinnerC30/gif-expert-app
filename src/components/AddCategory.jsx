import { useState } from "react";

export const AddCategory = () => {
  const [inputValue, setInputValue] = useState("Goku");
  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };
  const onSubmit = (event) => {
    event.preventDefault();
  };
  return (
    <>
      <form action="" onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="Buscar gifs"
          value={inputValue}
          onChange={onInputChange}
        />
      </form>
    </>
  );
};
