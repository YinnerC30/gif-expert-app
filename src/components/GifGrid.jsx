import { getGifs } from "../helpers/getGifs";
import { GifItem } from "./GifItem";
import { useState, useEffect } from "react";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);
  const getImages = async () => {
    const newImages = await getGifs(category);
    setImages(newImages);
  };

  useEffect(() => {
    getImages();
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-list">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
