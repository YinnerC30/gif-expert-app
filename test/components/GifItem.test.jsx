import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas del componente GifItem", () => {
  test("Toma de Snapshot", () => {
    const title = "Dragon Ball GIF by Toei Animation";
    const url =
      "https://media2.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=e52640fbrd5a6w92knd1zoivuwja3wyhb6mmkul1a0m1hiwg&ep=v1_gifs_search&rid=giphy.gif&ct=g";
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
});
