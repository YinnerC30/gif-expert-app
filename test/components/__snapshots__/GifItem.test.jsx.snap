// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifItem Toma de Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball GIF by Toei Animation"
      src="https://media2.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=e52640fbrd5a6w92knd1zoivuwja3wyhb6mmkul1a0m1hiwg&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Dragon Ball GIF by Toei Animation
    </p>
  </div>
</div>
`;
